### PR TITLE
on_os_block: allow keg_only

### PR DESCRIPTION
**Pull request by iMichka**
Needed for https://github.com/Homebrew/homebrew-core/pull/71854 for example

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Review by dawidd6 — APPROVED** [review]

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
I think `keg_only :shadowed_by_macos, ...` is a better fit here.

File: Library/Homebrew/rubocops/components_order.rb [hunk 1]
```diff
@@ -123,8 +131,20 @@ def check_on_os_block_content(component_precedence_list, on_os_block)
 
             valid_node ||= on_os_allowed_methods.include? child.method_name.to_s
 
+            shadowed = false
+            if child.method_name.to_s == "keg_only"
+              child.arguments.each do |arg|
```
Comment (dawidd6) [inline Library/Homebrew/rubocops/components_order.rb:136]:
```suggestion
              shadowed = child.arguments.find do |arg|
```

Comment (dawidd6) [reply Library/Homebrew/rubocops/components_order.rb:136]: I _think_ it should work and be simpler.

Comment (iMichka) [reply Library/Homebrew/rubocops/components_order.rb]: It works

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
I don't think this PR is required at all. The use-case in https://github.com/Homebrew/homebrew-core/pull/71854 does not require this change.

**Comment by iMichka** [comment]
Right, makes sense. I'll close then.

**Pull request closed without merging**